src/mercury_engine_data_structures/formats/bcskla.py

```diff
@@ -22,50 +20,47 @@
     this,
 )
 
-from mercury_engine_data_structures.adapters.enum_adapter import EnumAdapter
 from mercury_engine_data_structures.common_types import Float, VersionAdapter
 from mercury_engine_data_structures.construct_extensions.alignment import AlignTo
 from mercury_engine_data_structures.formats.base_resource import BaseResource
 from mercury_engine_data_structures.formats.property_enum import PropertyEnumDoubleUnsafe
 from mercury_engine_data_structures.game_check import Game
 
 
-class TimingTypeEnum(Enum):
-    ONE_BYTE = 8
-    TWO_BYTE = 0
-
-
 class DreadKFVAdapter(Adapter):
+    class TimingType:
+        ONE_BYTE = 8
+        TWO_BYTE = 0
+
     SUBCON = Struct(
-        timing_type=EnumAdapter(TimingTypeEnum, Int16ul),
+        timing_type=Int16ul,
         count=Int16ul,
-        timings=IfThenElse(this.timing_type == TimingTypeEnum.TWO_BYTE, Int16ul[this.count], Int8ul[this.count]),
+        timings=IfThenElse(this.timing_type == TimingType.TWO_BYTE, Int16ul[this.count], Int8ul[this.count]),
         _padding=AlignTo(4, b"\xff"),
-        values=Array(this.count, Struct(value=Float, derivative=Float)),
-    )
+        values=Array(this.count, Float[2]),
+    ).compile()
 
     def __init__(self):
         super().__init__(self.SUBCON)
 
     def _decode(self, obj, context, path):
-        res = ListContainer()
-        for i in range(obj.count):
-            res.append(
-                Container(
-                    time=obj.timings[i],
-                    value=obj["values"][i].value,
-                    derivative=obj["values"][i].derivative,
-                )
+        return ListContainer(
+            Container(
+                time=time,
+                value=v[0],
+                derivative=v[1],
             )
-
-        return res
+            for time, v in zip(obj.timings, obj["values"], strict=True)
```

Comment:
probably should make it access timings and values as both properties or both keys.

Reply:
not following

Reply:
like object["timings"], obj["values"] or obj.timings, obj.values